src/postgraphql/graphiql/graphiql.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import { render } from 'react-dom'
+import GraphiQL from 'graphiql'
+import { buildClientSchema, introspectionQuery } from 'graphql'
+import 'graphiql/graphiql.css'
+
+class EnhancedGraphiQL extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      schema: undefined,
+      token: undefined,
+    }
+  }
+
+  componentWillMount() {
+    setupStream({
+      onChange: this.onChange,
+      onOpen: this.onOpen,
+      onError: this.onError,
+    })
+  }
+
+  render() {
+    return (
+      <GraphiQL schema={this.state.schema} fetcher={this.fetcher}>
+        <GraphiQL.Logo>PostGraphQL GraphiQL</GraphiQL.Logo>
+        <GraphiQL.Footer>
+          <input onBlur={this.onTokenSet}/>
+        </GraphiQL.Footer>
+      </GraphiQL>
+    )
+  }
+
+  onTokenSet = event => {
+    this.setState({ token: event.target.value })
+  }
+
+  onChange = event => {
+    this.fetcher({ query: introspectionQuery })
```

Comment:
Should we `this.setState({ schema: null })` here while we are fetching the new schema? This way GraphiQL won’t display the old one.